src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.cs

```diff
@@ -133,6 +133,34 @@ public async Task NoDataIsLostWhenWritingToFile(ReadWriteMode mode)
             byte[] allBytes = File.ReadAllBytes(filePath);
             Assert.Equal(writtenBytes.ToArray(), allBytes);
         }
+
+        [Theory]
+        [InlineData(FileAccess.Write)]
+        [InlineData(FileAccess.ReadWrite)] // FileAccess.Read does not allow for length manipulations
+        public async Task LengthIsNotCachedAfterHandleHasBeenExposed(FileAccess fileAccess)
+        {
+            using FileStream stream = (FileStream)await CreateStream(Array.Empty<byte>(), fileAccess);
+            using FileStream createdFromHandle = new FileStream(stream.SafeFileHandle, fileAccess);
```

Comment:
Do you also need a test that marks it as exposed part way through rather than immediately?

Reply:
I don't think so. In this test, we have a `FileStream` that gets exposed by accessing `SafeFileHandle` and another one that gets exposed by creating it from the handle. The modification of `Length` is verified in both directions. IMO it's enough.

If you believe that there is a good reason to do it, please let me know why and I'll be happy to send another PR. In the meantime, I am going to merge it as I really want to start testing other repos against the new implementation.

Reply:
Because you added code that ignores a _length cache if _exposed is true.  For both stream and createFromHandle here, you'll never get to the point where you cache _length in Length, because _exposed is true before Length is ever accessed.